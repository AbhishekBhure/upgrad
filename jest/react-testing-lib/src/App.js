import React, { useState, useEffect } from "react";
import "./index.css";
import ListItem from "./components/ListItem";
import AddToDo from "./components/AddToDo";

import { getToDos, setDone, deleteToDo, addTodo } from "./services";

const App = () => {
  const [todos, setToDos] = useState();
  useEffect(() => {
    getToDos().then(data => setToDos(data));
  }, []);
  const onDoneHandler = (status, id) => {
    setDone(status, id).then(todo => {
      setToDos(
        todos.map(i =>
          i.id === id
            ? {
                ...i,
                done: todo.done
              }
            : i
        )
      );
    });
  };
  const onDeleteHandler = id => {
    deleteToDo(id).then(() => {
      setToDos(todos.filter(i => i.id !== id));
    });
  };
  const onAddHandler = task => {
    addTodo(task).then(todo => setToDos([...todos, todo]));
  };
  return (
    <div className="app">
      <div className="title">Tasks</div>
      <AddToDo onAdd={onAddHandler} />
      <div className="list">
        {todos &&
          todos.map(({ id, title, done }) => (
            <ListItem
              id={id}
              done={done}
              title={title}
              key={id}
              onDone={onDoneHandler}
              onDelete={onDeleteHandler}
            />
          ))}
      </div>
    </div>
  );
};

App.defaultProps = {
  init: []
};

export default App;
