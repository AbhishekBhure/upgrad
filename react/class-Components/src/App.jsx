import React from "react";
import Timer from "./components/Timer";
import Child from "./components/Child";

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      count: 0,
    };

    this.state = {
      showTime: true,
    };
  }

  increment = () => {
    this.setState({ count: this.state.count + 1 });
  };

  decrement = () => {
    if (this.state.count > 0) {
      this.setState({ count: this.state.count - 1 });
    }
  };

  onSwitchClock = () => {
    this.setState({ showTime: !this.state.showTime });
  };

  render() {
    return (
      <>
        <h1>Hello</h1>
        <button onClick={this.decrement}> - </button>

        <Child count={this.state.count} />
        {/* <p>{this.state.count}</p> */}
        <button onClick={this.increment}>+</button>

        <div className="flex flex-col justify-center">
          <button
            onClick={this.onSwitchClock}
            type="button"
            className="bg-white w-[50px] m-auto rounded text-black mb-4"
          >
            {this.state.showTime ? "Hide" : "Show"}
          </button>
          {this.state.showTime && <Timer toggle={this.state.showTime} />}
        </div>
      </>
    );
  }
}

export default App;
