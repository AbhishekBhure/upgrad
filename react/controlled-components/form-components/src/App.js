import React from "react";
import MessageBox from "./MessageBox";

const App = () => (
  <MessageBox nameVal="Abhi" natureVal="Tech Support" queryVal="I need Help" />
);

export default App;
