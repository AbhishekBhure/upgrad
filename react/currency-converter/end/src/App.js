import React, {Component} from "react";
import CurrencyInput from "./CurrencyInput";
import openRates from "./services/openRates";

class App extends Component {
  state = {
    from: "USD",
    to: "INR",
    rate: 1,
    fromAmt: 1,
    toAmt: 1
  };
  componentDidMount() {
    this.fetchRates(this.state.from, this.state.to);
  }
  componentDidUpdate(_, prevState) {
    if (prevState.from !== this.state.from || prevState.to !== this.state.to) {
      this.fetchRates(this.state.from, this.state.to);
    }
  }
  fetchRates = async (base = "USD", symbol = "INR") => {
    const {rate} = await openRates(base, symbol);
    this.setState({
      rate
    });
  };
  computeResult = key => {
    let {fromAmt, toAmt} = this.state;

    if (fromAmt !== null) {
      toAmt = parseFloat(fromAmt * this.state.rate).toFixed(2);
    } else {
      fromAmt = parseFloat(toAmt / this.state.rate).toFixed(2);
    }
    return key === "from" ? fromAmt : toAmt;
  };
  setAmount = (amt, field) => {
    if (field === "from") {
      this.setState({
        fromAmt: amt,
        toAmt: null
      });
    } else {
      this.setState({
        fromAmt: null,
        toAmt: amt
      });
    }
  };
  render() {
    return (
      <div className="currency-converter">
        <CurrencyInput
          symbol={this.state.from}
          selectSymbol={sym => this.setState({from: sym})}
          setAmount={amt => this.setAmount(amt, "from")}
          amount={this.computeResult("from")}
        />
        <CurrencyInput
          symbol={this.state.to}
          selectSymbol={sym => this.setState({to: sym})}
          setAmount={amt => this.setAmount(amt, "to")}
          amount={this.computeResult("to")}
        />
      </div>
    );
  }
}

export default App;
