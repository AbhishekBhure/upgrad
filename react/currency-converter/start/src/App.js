import React, {Component} from "react";

class App extends Component {
  render() {
    return <div className="currency-converter"></div>;
  }
}

export default App;
