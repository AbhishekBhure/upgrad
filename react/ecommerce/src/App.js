import { Component } from "react";
import Navbar from "./Components/Navbar";
import Banner from "./Components/Banner";
import ProductItem from "./Components/ProductItem";
import Allproduct from "./Components/Allproduct";
import ScrollToTop from "./Components/ScrollToTop";

const dealProducts = [
  {
    id: 1,
    imgUrl: "",
    prodName: "iPhone 15",
    prodBrand: "iPhone",
    price: "$6500",
  },
  {
    id: 2,
    imgUrl: "",
    prodName: "Samsung A50",
    prodBrand: "Samsung",
    price: "$500",
  },
  {
    id: 3,
    imgUrl: "",
    prodName: "Motorola M53",
    prodBrand: "Motorola",
    price: "$200",
  },
  {
    id: 4,
    imgUrl: "",
    prodName: "Redmi Note8 Pro",
    prodBrand: "Redmi",
    price: "$6500",
  },
  {
    id: 5,
    imgUrl: "",
    prodName: "iPhone 12 Pro",
    prodBrand: "iPhone",
    price: "$900",
  },
];

const allProd = [
  {
    id: 1,
    imgUrl: "",
    prodName: "iPhone 15",
    prodBrand: "iPhone",
    price: "$6500",
  },
  {
    id: 2,
    imgUrl: "",
    prodName: "Samsung A50",
    prodBrand: "Samsung",
    price: "$500",
  },
  {
    id: 3,
    imgUrl: "",
    prodName: "Motorola M53",
    prodBrand: "Motorola",
    price: "$200",
  },
  {
    id: 4,
    imgUrl: "",
    prodName: "Redmi Note8 Pro",
    prodBrand: "Redmi",
    price: "$6500",
  },
  {
    id: 5,
    imgUrl: "",
    prodName: "iPhone 12 Pro",
    prodBrand: "iPhone",
    price: "$900",
  },
  {
    id: 6,
    imgUrl: "",
    prodName: "iPhone 15",
    prodBrand: "iPhone",
    price: "$6500",
  },
  {
    id: 7,
    imgUrl: "",
    prodName: "Samsung A50",
    prodBrand: "Samsung",
    price: "$500",
  },
  {
    id: 8,
    imgUrl: "",
    prodName: "Motorola M53",
    prodBrand: "Motorola",
    price: "$200",
  },
  {
    id: 9,
    imgUrl: "",
    prodName: "Redmi Note8 Pro",
    prodBrand: "Redmi",
    price: "$6500",
  },
  {
    id: 10,
    imgUrl: "",
    prodName: "iPhone 12 Pro",
    prodBrand: "iPhone",
    price: "$900",
  },
];

class App extends Component {
  state = { cartCount: 0 };

  onAddToCart = () => {
    this.setState((prev) => ({
      cartCount: prev.cartCount + 1,
    }));
  };
  render() {
    const { cartCount } = this.state;

    return (
      <div>
        <div>
          <Navbar cartCount={cartCount} />
          <Banner />
        </div>
        <div>
          <h2 className="text-center text-xl font-bold">Deals Of the Day</h2>
          <ul className="flex flex-wrap gap-5 justify-evenly">
            {dealProducts.map((product) => (
              <ProductItem
                product={product}
                key={product.id}
                onAddToCart={this.onAddToCart}
              />
            ))}
          </ul>
        </div>
        <div className="w-full">
          <h2 className="text-center text-xl font-bold">All Products</h2>
          <ul className="flex flex-wrap w-full gap-5 justify-evenly ">
            {allProd.map((allproduct) => (
              <Allproduct allproduct={allproduct} key={allproduct.id} />
            ))}
          </ul>
        </div>
        <ScrollToTop />
      </div>
    );
  }
}

export default App;
