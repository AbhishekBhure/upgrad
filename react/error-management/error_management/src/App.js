import React, {useState, useEffect} from "react";
import getProducts from "./services/getProducts";
import Product from "./components/Product";
import ErrorManager from "./ErrorManager";

const App = () => {
  const [products, setProducts] = useState([]);
  const [category, setCategory] = useState("");
  useEffect(() => {
    getProducts().then(products => setProducts(products));
  });
  const filterByCategory = cat => {
    category === cat ? setCategory("") : setCategory(cat);
  };
  return (
    <div className="container">
      <ErrorManager onClose={() => setCategory("")}>
        <div className="categories">
          {[...new Set(products.map(p => p.type))].map(cat => (
            <div
              className={category === cat ? "cat-btn cat-btn-on" : "cat-btn"}
              key={cat}
              onClick={() => filterByCategory(cat)}
            >
              {cat}
            </div>
          ))}
        </div>

        {category ? (
          products
            .filter(p => p.type === category)
            .map(p => <Product item={p} key={p.id} />)
        ) : (
          <p className="pick-text">Pick a category</p>
        )}
      </ErrorManager>
    </div>
  );
};

export default App;
