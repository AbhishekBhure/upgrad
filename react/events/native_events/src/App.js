import React, { useState } from "react";

const App = () => {
  const [xPos, setXPos] = useState(0);
  const [yPos, setYPos] = useState(0);
  const [marker, setMarker] = useState([]);

  const onMouseMoveHandler = (e) => {
    setXPos(e.nativeEvent.offsetX);
    setYPos(e.nativeEvent.offsetY);
  };

  const onClickHandler = (e) => {
    setMarker([
      ...marker,
      {
        id: new Date().getTime(),
        x: e.nativeEvent.offsetX,
        y: e.nativeEvent.offsetY,
      },
    ]);
  };
  return (
    <div className="container">
      <div className="stats">
        {xPos}, {yPos}
        <button id="clear-btn" onClick={() => setMarker([])}>
          Clear
        </button>
      </div>
      <div
        className="canvas"
        onMouseMove={onMouseMoveHandler}
        onClick={onClickHandler}
      >
        {marker.map((m) => (
          <div key={m.id} className="dot" style={{ left: m.x, top: m.y }} />
        ))}
      </div>
    </div>
  );
};

export default App;
