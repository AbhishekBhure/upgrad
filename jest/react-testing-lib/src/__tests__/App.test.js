import React from "react";
import { wait, render, fireEvent, cleanup } from "@testing-library/react";

jest.mock("../services", () => {
  return {
    getToDos: () =>
      Promise.resolve([
        {
          title: "Book concert tickets",
          done: false,
          id: 1
        },
        {
          title: "Buy a new iPhone",
          done: true,
          id: 2
        },
        {
          title: "Write an article on React",
          done: true,
          id: 3
        }
      ]),
    addTodo: obj => Promise.resolve({ ...obj, id: 4 })
  };
});

import App from "../App";

afterEach(cleanup);

describe("Testing the App", () => {
  it("Renders successfully", async () => {
    const { asFragment, getByPlaceholderText } = render(<App />);
    // Render a basic snapshot
    await wait(() => expect(asFragment()).toMatchSnapshot());

    // Add a task and take a snapshot
    const taskInput = getByPlaceholderText(/Add a task/i);
    fireEvent.change(taskInput, {
      target: { value: "This is a brand new task" }
    });
    fireEvent.keyUp(taskInput, { key: "Enter", keyCode: 13 });
    await wait(() => expect(asFragment()).toMatchSnapshot());
  });
});
