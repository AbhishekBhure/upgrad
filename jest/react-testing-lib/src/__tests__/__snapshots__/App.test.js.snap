// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App Renders successfully 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="title"
    >
      Tasks
    </div>
    <div
      class="add-todo"
    >
      <input
        placeholder="Add a task"
        type="text"
        value=""
      />
    </div>
    <div
      class="list"
    >
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Book concert tickets
        </div>
        <div
          class="status-btn"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Buy a new iPhone
        </div>
        <div
          class="status-btn status-done"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Write an article on React
        </div>
        <div
          class="status-btn status-done"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Testing the App Renders successfully 2`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="title"
    >
      Tasks
    </div>
    <div
      class="add-todo"
    >
      <input
        placeholder="Add a task"
        type="text"
        value=""
      />
    </div>
    <div
      class="list"
    >
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Book concert tickets
        </div>
        <div
          class="status-btn"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Buy a new iPhone
        </div>
        <div
          class="status-btn status-done"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          Write an article on React
        </div>
        <div
          class="status-btn status-done"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
      <div
        class="list-item"
      >
        <div
          class="task"
        >
          This is a brand new task
        </div>
        <div
          class="status-btn"
        />
        <div
          class="delete-btn"
        >
          X
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
